src/hawk/collision.lua

```diff
@@ -222,7 +223,7 @@ function module.move_x(map, player, x, y, width, height, dx, dy)
         
           if new_x <= tile_x and tile_x <= x then
             if player.wall_pushback then
-              player:wall_pushback()
+              player:wall_pushback(tile)
```

Comment:
The only reason we pass `tile` as an argument here is to return on the function before it runs. I think it's cleaner to move that condition up here before we even call that function. Thoughts?